crates/pop-contracts/src/utils/metadata.rs

```diff
@@ -0,0 +1,367 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::errors::Error;
+use contract_extrinsics::ContractArtifacts;
+use contract_transcode::ink_metadata::MessageParamSpec;
+use scale_info::form::PortableForm;
+use std::path::Path;
+
+#[derive(Clone, PartialEq, Eq)]
```

Comment:
Personal preference, but I find having the doc comments before the attributes is nicer, as it keeps the code grouped together with type declaration and it makes the doc comments the first thing you read.